dev-client/src/screens/SoilScreen/TextureGuideScreen.tsx

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright © 2023 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {Text, ScrollView, Column, Button, Box} from 'native-base';
+import {SoilPitInputScreenProps} from 'terraso-mobile-client/screens/SoilScreen/components/SoilPitInputScreenScaffold';
+import {ScreenScaffold} from 'terraso-mobile-client/screens/ScreenScaffold';
+import {AppBar} from 'terraso-mobile-client/navigation/components/AppBar';
+import {useTranslation} from 'react-i18next';
+import {RadioBlock} from 'terraso-mobile-client/components/RadioBlock';
+import {useMemo, useState} from 'react';
+import {
+  SoilTexture,
+  updateDepthDependentSoilData,
+} from 'terraso-client-shared/soilId/soilIdSlice';
+import {Icon} from 'terraso-mobile-client/components/Icons';
+import {useDispatch} from 'terraso-mobile-client/store';
+import {useNavigation} from 'terraso-mobile-client/navigation/hooks/useNavigation';
+import {ResizeMode, Video} from 'expo-av';
+import {Image, StyleSheet, View} from 'react-native';
+
+const LENGTH_IMAGE = require('terraso-mobile-client/assets/texture/guide/length.png');
+
+export const TextureGuideScreen = (props?: SoilPitInputScreenProps) => {
+  const {t} = useTranslation();
+  const [ball, setBall] = useState<'YES' | 'NO'>();
+  const [ribbon, setRibbon] = useState<'YES' | 'NO'>();
+  const [ribbonLength, setRibbonLength] = useState<'SM' | 'MD' | 'LG'>();
+  const [grit, setGrit] = useState<'GRITTY' | 'SMOOTH' | 'NEITHER'>();
+
+  const result = useMemo<SoilTexture | undefined>(() => {
+    /* eslint-disable curly */
```

Comment:
question: why don't we want curly braces here?

Reply:
it makes the logic flow much harder to parse since return values are no longer on the same line as their condition and the whole block takes up 3 times as many lines